node-js-basics-101/node_modules/uws/uws.js

```diff
@@ -0,0 +1,563 @@
+    }
+
+    set onopen(f) {
+        if (f) {
+            this.internalOnOpen = f;
+        } else {
+            this.internalOnOpen = noop;
+        }
+    }
+
+    set onclose(f) {
+        if (f) {
+            this.internalOnClose = (code, message) => {
+                f({code: code, reason: message});
+            };
+        } else {
+            this.internalOnClose = noop;
+        }
+    }
+
+    set onerror(f) {
+        if (f && this instanceof WebSocketClient) {
+            this.internalOnError = f;
+        } else {
+            this.internalOnError = noop;
+        }
+    }
+
+    emit(eventName, arg1, arg2) {
+        if (eventName === 'message') {
+            this.internalOnMessage(arg1);
+        } else if (eventName === 'close') {
+            this.internalOnClose(arg1, arg2);
+        } else if (eventName === 'ping') {
+            this.onping(arg1);
+        } else if (eventName === 'pong') {
+            this.onpong(arg1);
+        }
+        return this;
+    }
+
+    on(eventName, f) {
+        if (eventName === 'message') {
+            if (this.internalOnMessage !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnMessage = f;
+        } else if (eventName === 'close') {
+            if (this.internalOnClose !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnClose = f;
+        } else if (eventName === 'ping') {
+            if (this.onping !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.onping = f;
+        } else if (eventName === 'pong') {
+            if (this.onpong !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.onpong = f;
+        } else if (eventName === 'open') {
+            if (this.internalOnOpen !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnOpen = f;
+        } else if (eventName === 'error' && this instanceof WebSocketClient) {
+            if (this.internalOnError !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnError = f;
+        }
+        return this;
+    }
+
+    once(eventName, f) {
+        if (eventName === 'message') {
+            if (this.internalOnMessage !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnMessage = (message) => {
+                this.internalOnMessage = noop;
+                f(message);
+            };
+        } else if (eventName === 'close') {
+            if (this.internalOnClose !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnClose = (code, message) => {
+                this.internalOnClose = noop;
+                f(code, message);
+            };
+        } else if (eventName === 'ping') {
+            if (this.onping !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.onping = () => {
```

Comment:
__[jshint]__
* 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

<!-- Comment by [SideCI](https://sideci.com) -->